sdk/digitaltwins/Azure.DigitalTwins.Core/tests/PublishTelemetryTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.DigitalTwins.Core.Models;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Azure.DigitalTwins.Core.Tests
+{
+    /// <summary>
+    /// Tests for DigitalTwinsClient APIs that handle publishing telemetry messages to Azure Digital Twins.
+    /// </summary>
+    public class PublishTelemetryTests : E2eTestBase
+    {
+        public PublishTelemetryTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        // Infrastructure setup script uses this hardcoded value when linking the test eventhub to the test digital twins instance.
+        private const string EndpointName = "someEventHubEndpoint";
+
+        [Test]
+        public async Task PublishTelemetry_Lifecycle()
+        {
+            // Setup
+
+            // Create a DigitalTwinsClient instance.
+            DigitalTwinsClient client = GetClient();
+
+            string wifiComponentName = "wifiAccessPoint";
+            string wifiModelId = await GetUniqueModelIdAsync(client, TestAssetSettings.WifiModelIdPrefix).ConfigureAwait(false);
+            string roomWithWifiModelId = await GetUniqueModelIdAsync(client, TestAssetSettings.RoomWithWifiModelIdPrefix).ConfigureAwait(false);
+            string roomWithWifiTwinId = await GetUniqueTwinIdAsync(client, TestAssetSettings.RoomWithWifiTwinIdPrefix).ConfigureAwait(false);
+            string eventRouteId = $"someEventRouteId-{GetRandom()}";
+
+            try
+            {
+                // Create an event route for the digital twins client.
+                EventRoute eventRoute = await CreateEventRoute(client, eventRouteId);
+
+                // Create the models needed for the digital twin.
+                await CreateModelsAndTwins(client, wifiModelId, roomWithWifiModelId, wifiComponentName, roomWithWifiTwinId);
+
+                // Act - Test publishing telemetry to a digital twin.
+                Response publishTelemetryResponse = await client.PublishTelemetryAsync(roomWithWifiTwinId, "{\"Telemetry1\": 5}");
+
+                // Assert
+                publishTelemetryResponse.Status.Should().Be((int)HttpStatusCode.NoContent);
```

Comment:
Why is the response NoContent and not Ok? Any idea?

Reply:
This API is a fire and forget API. It returns a 204, which is NoContent.

Reply:
Swagger: https://github.com/Azure/azure-sdk-for-net/blob/b7bdeb9c1d6b5b6642b23e467edad1a0eb43f59d/sdk/digitaltwins/Azure.DigitalTwins.Core/src/swagger/digitaltwins.json#L820